### DISCUSSION TITLE
HtmlText not save

### DISCUSSION THREAD

**Discussion by mortezashoghi** [oqtane/oqtane.framework#1952] [Q&A]
Hi

Changes are not saved in the Htmltext module

I did the following steps: 
1- download v3.0.2  
2- build  with Vs2022
3-everything works good when i  execute  that without debuging .

** This problem occurs after the project is published.

**Comment by leigh-pointer** [comment]
@mortezashoghi when you say publish how did you publish?  Was is using the package scripts that are in the framework solution?

**Reply by mortezashoghi** [reply]
Thanks for reply. 

**How should i use the package scripts that are in the framework solution?**

**Reply by leigh-pointer** [reply]
@mortezashoghi 
1. In the framework folder Oqtane.Framework you will find a folder  Oqtane.Package open this and you will find a file call **release.cmd** you must run this (It needs to be run with admin rights - right click on file and run as administer)  the build will take a moment or two.
2.  once it is is complete you will have a few files created in the Oqtane.Package folder.  The Oqtane.Framework.3.0.2.Install.zip file contains the complete compiled solution.

**Reply by mortezashoghi** [reply]
Ok I did this two  steps. But the  HtmlText module save button wont work, And changes content in this module not saved.

**Reply by leigh-pointer** [reply]
Are you running is IIS ?
Is webDAV disabled? webDAV must be disabled otherwise save will not work.

The webDAV  settings can be modified in IIS or in the web.config

In you web.config (updating this is easier and quicker) update this section:
```

    <system.webServer>
      <modules>
        <remove name="WebDAVModule" />
      </modules>
      <handlers>
        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
        <remove name="WebDAV" />
      </handlers>
      <aspNetCore processPath="dotnet" arguments=".\Oqtane.Server.dll" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" hostingModel="inprocess" />
    </system.webServer>
```

**Reply by mortezashoghi** [reply]
YES

well done

thanks

**Reply by leigh-pointer** [reply]
@mortezashoghi brilliant :) can you marked this a answered.

@sbwalker disabling the webDAV, is this not something we could add to the web.config ?

**Reply by sbwalker** [reply]
Disabling WebDAV is mentioned in step #1 of the IIS installation instructions https://www.oqtane.org/Resources/Blog/PostId/542/installing-oqtane-on-iis - however I do think it is a good idea to include it in the web.config. The question is how to do it, as web.config is added during the publish stage of a release (there is no web.config file in the repo currently).

**Reply by leigh-pointer** [reply]
@sbwalker what would happen if we added the web.config to the solution and then built the packages, would it not include what we have set instead of the autogenerated one?

**Reply by ADefWebserver** [reply]
Including the web.config will likely break the Azure deployments. My recommendation is to instruct people how to change the settings in ApplicationHost.comfig
![image](https://user-images.githubusercontent.com/1857799/150375524-4596dd69-d71d-4daa-b2ee-7685d65096a2.png)

**Reply by leigh-pointer** [reply]
@ADefWebserver the web.config is already deployed in the install package.  What I should have said is, is there a way a include the modifications for the webDAV removal in the web.config

**Reply by ADefWebserver** [reply]
@leigh-pointer Of course you are correct. I was thinking of how I messed up my Azure deployment when I manually created a Web.Config when I wrote: [Hosting A Blazor Visual Studio Project Directly In IIS](https://blazorhelpwebsite.com/ViewBlogPost/56). 
What I should have said was "please make sure the setting won't be a problem when the Web.config is run in Azure" 😁

**Reply by Rodien** [reply]
@leigh-pointer This is possible take a look at my pull request https://github.com/oqtane/oqtane.framework/pull/1957 > @ADefWebserver the web.config is already deployed in the install package.  What I should have said is, is there a way a include the modifications for the webDAV removal in the web.config
